server/monitor/src/main/java/org/apache/accumulo/monitor/Monitor.java

```diff
@@ -484,6 +485,17 @@ public void run() {
       }
     }).start();
 
+    Threads.createThread("Compaction fetcher", () -> {
+      while (true) {
+        try {
+          fetchCompactions();
+        } catch (Exception e) {
+          log.warn("{}", e.getMessage(), e);
+        }
+        sleepUninterruptibly(5, TimeUnit.SECONDS);
+      }
+    }).start();
+
```

Comment:
The problem with this pattern is that there's a constant background thread doing RPC calls to the tservers to collect these stats, even if nobody is accessing the page on the monitor. And, it has to talk to every tserver, of which there could be thousands. This is a lot of constant activity that might not even be needed. The general pattern of the monitor is to avoid refreshing data if nobody is accessing the page, and to limit the refresh of the data shown on the page to collect it no more frequently than once every 5 seconds. This is how we collect the "mmi" object, anyway.

Fast display should not bog down the cluster like this.

Reply:
> The problem with this pattern is that there's a constant background thread doing RPC calls to the tservers to collect these stats, even if nobody is accessing the page on the monitor. And, it has to talk to every tserver, of which there could be thousands. This is a lot of constant activity that might not even be needed. The general pattern of the monitor is to avoid refreshing data if nobody is accessing the page, and to limit the refresh of the data shown on the page to collect it no more frequently than once every 5 seconds. This is how we collect the "mmi" object, anyway.
> 
> Fast display should not bog down the cluster like this.

This is also currently being done for active scans. So if it's an issue with the fetch compactions thread, we should do something with the scans fetch as well.

Reply:
> This is also currently being done for active scans. So if it's an issue with the fetch compactions thread, we should do something with the scans fetch as well.

Yeah, I saw that. It seems like it's a problem there, too. I'm not sure when that got added, but we definitely don't need the monitor constantly talking to every server in the cluster every 5 seconds, if the monitor doesn't even have any active clients requesting pages.

Reply:
I am working on the idea to have a button to fetch the info. I like the manual fetch in this case. We can still restrict it to only fetch every minute so an Admin can't spam the tservers. This way if no one cares, there won't ever be any wasted work. If a user does care, they will have the info provided on the page through an AJAX call.